### PR TITLE
⚡️ Fast browser

### PR DESCRIPTION
**Pull request by christianbaroni**
## What changed (plus any additional context for devs)
- Converts our `Input` component to TS
- Eliminates almost all performance bottlenecks in the browser
- Heavily cuts down on the usage of JS state in the browser and in the browser context, outside of tabStates and a bit of remaining reliance on the JS-side `activeTabIndex`
- Converts toggleTabView to `toggleTabViewWorklet`, which is now only called from the UI thread — result is very responsive entering/exiting of the tab view
- Fixes tab closing logic and a crash that occurred when the last tab was closed
- Adds a new tab button to the tab view, along with a button to exit the tab view
- Adds a `uniqueId` to each newly created BrowserTab to better identify tabs, which also now serves as the BrowserTab's key, eliminating unnecessary recreations of the tabs when only a tab's index has changed
- Adds a swipe-to-close gesture within the tab view
- Adds the [faster-image](https://github.com/candlefinance/faster-image) package, which seems to be the only(?) RN image component that allows its `source` prop to be set from a reanimated UI thread context / via useAnimatedProps
- Adds an `enableFasterImage` option to `ImgixImage` (disabled by default, currently used only in the browser `Homepage` component)
- Fixes screenshots and adds screenshot pruning
- Fixes URL vs search detection with a more reliable regex check
- Switches from using a controlled Input to an uncontrolled one, which eliminates a ton of re-renders and input jank, and prevents redirect loops caused by the URL state changes
- Now possible to navigate within single-page web apps
- Fixes browser search bar font styles on Android
- Created `animationConfigs.ts` to serve as a source of truth for reanimated animation configs — it exports both `SPRING_CONFIGS` and `TIMING_CONFIGS`
- Misc. UI polish, style fixes

## Notes
- There are a couple of remaining touch event bugs that result in:
  - The tab close button not picking up presses (swiping to dismiss works)
  - The `Homepage` component not being scrollable
- The calling of `pruneScreenshots()` could be better managed — we may want to put a frequency cap on it, and/or defer its execution
- The tabStates data is currently being persisted with useMMKVObject — we'll probably end up switching that up, but it was necessary to persist tabStates for the uniqueId solution to work
- Triggering closeTab multiple times in quick succession sometimes prevents a tab from successfully closing — will likely be fixed by using zustand to manipulate tabStates
- [x] ~TODO: Catch up with latest develop~

## Screen recordings / screenshots

https://github.com/rainbow-me/rainbow/assets/7061887/b33b76ee-f80e-471b-8751-6f9d2ace2e84

## What to test
- Android

### CODE REVIEW

File: src/components/DappBrowser/search/SearchBar.tsx [hunk 1]
```diff
@@ -1,49 +1,142 @@
...
-    ],
-  }));
+  const isFocusedValue = useSharedValue(false);
+  const [isFocused, setIsFocused] = useState<boolean>(false);
+
+  const keyboardHeight = useKeyboardHeight({ shouldListen: isFocused });
+
+  const inputRef = useAnimatedRef<TextInput>();
+
+  const tabId = useMemo(() => tabStates?.[activeTabIndex]?.uniqueId, [activeTabIndex, tabStates]);
```
Comment (benisgold) [inline src/components/DappBrowser/search/SearchBar.tsx:37]: why do these two need to be memoized?

**Review by brunobar79 — APPROVED** [review]
🔥 🔥 🔥

File: src/components/DappBrowser/search/SearchBar.tsx (hunk 1 above)
Comment (christianbaroni) [reply src/components/DappBrowser/search/SearchBar.tsx]: Good call, fixed and removed similar unnecessary memoization in `BrowserTab`

**Review by benisgold — APPROVED** [review]

**Review by skylarbarrera — APPROVED** [review]
added some tix for the TODOs, comments are more notes for future work

**Comment by jxom** [comment]
It's time to ditch Chrome.

**Pull request merged**